### PR TITLE
ECF-78 Implement logout

### PR DESCRIPTION
**Pull request by amazimbe**
### Context
Implement user logout

### Changes proposed in this pull request

Code changes to enable users to log out

### CODE REVIEW

**Review by ethanmills — APPROVED** [review]

File: app/views/layouts/application.html.erb [hunk 1]
```diff
@@ -45,12 +45,15 @@
           <button type="button" role="button" data-module="govuk-button" class="govuk-header__menu-button govuk-js-header-toggle" aria-controls="navigation" aria-label="Show or hide Top Level Navigation">Menu</button>
           <nav>
             <ul id="navigation" class="govuk-header__navigation " aria-label="Top Level Navigation">
-              <li class="govuk-header__navigation-item govuk-header__navigation-item--active">
-                <%= link_to "Navigation item 1", "#", class: "govuk-header__link" %>
-              </li>
+              <% if user_signed_in? %>
               <li class="govuk-header__navigation-item">
-                <%= link_to "Navigation item 2", "#", class: "govuk-header__link" %>
+                <%= link_to "Logout", destroy_user_session_path, method: "delete", class: "govuk-header__link" %>
```
Comment (kicferk1) [inline app/views/layouts/application.html.erb:50]: So we `delete` here, why do we have `config.sign_out_via = :get` line in our config? And in our tests?

Also, I would argue that `post` or `delete` are in fact much better than `get` to represent a sign out action.

Comment (amazimbe) [reply app/views/layouts/application.html.erb:50]: Delete is definitely preferable but very difficult to use from an anchor unless you are handling the anchor with an Ajax call. method: "delete" here just adds data-method=delete to the link and by itself doesn't do anything. As I have no desire to handle this link with ajax, I should have taken it out but forgot.

Comment (amazimbe) [reply app/views/layouts/application.html.erb:50]: I have taken it out

Comment (kicferk1) [reply app/views/layouts/application.html.erb]: Sorry, I don't quite understand, what is the behaviour that's difficult to get exactly? 

I really don't like the idea of signing out with `get`. Signing out doesn't seem like an idempotent operation for me for one.

And I had a look at teaching vacancies, and apply for teacher training - they both use navigation items with delete for logout.

**Review by kicferk1 — APPROVED** [review]

**Pull request merged**